packages/handlersjs-core/lib/handlers/waterfall.handler.ts

```diff
@@ -36,20 +41,14 @@ export class WaterfallHandler<T, S> extends Handler<T, S> {
     input: T,
     intermediateOutput: S,
     handlers: Handler<T, S>[],
-  ): Observable<Handler<T, S>> {
-
-    if (!this.handlers) {
-
-      throw new HandlerArgumentError('Argument this.handlers should be set.', this.handlers);
-
-    }
+  ): Observable<Handler<T, S> | undefined> {
 
     return from(handlers).pipe(
       switchMap((handler) =>
         handler.canHandle(input, intermediateOutput).pipe(map((canHandle) => ({ canHandle, handler })))),
       first((result) => result.canHandle, null),
       map((canHandle) => canHandle?.handler),
-      switchMap((handler) => handler ? of(handler) : throwError(new Error('Handler cannot be undefined.'))),
+      switchMap((handler) => of(handler)),
```

Comment:
This error was added in the TS strict mode PR but I know realized while testing this that it should be able to return an observable of undefined for the ternary, on line 33 above, to actually work.